internal/services/cognitive/azure_ai_services_resource.go

```diff
@@ -0,0 +1,757 @@
+			if err != nil {
+				return fmt.Errorf("expanding `identity`: %+v", err)
+			}
+			props.Identity = expandIdentity
+
+			future, err := client.AccountsCreate(ctx, id, props)
+			if err != nil {
+				return fmt.Errorf("creating %s: %+v", id, err)
+			}
+
+			if err := future.Poller.PollUntilDone(ctx); err != nil {
+				return fmt.Errorf("waiting for creating of %s: %+v", id, err)
+			}
+
+			customMangedKey, err := expandAzureAIServicesCustomerManagedKey(model.CustomerManagedKey)
+			if err != nil {
+				return fmt.Errorf("expanding `customer_managed_key`: %+v", err)
+			}
+
+			if customMangedKey != nil {
+				props.Properties.Encryption = customMangedKey
+				futureUpdate, err := client.AccountsUpdate(ctx, id, props)
+				if err != nil {
+					return fmt.Errorf("updating %s: %+v", id, err)
+				}
+
+				if err := futureUpdate.Poller.PollUntilDone(ctx); err != nil {
+					return fmt.Errorf("waiting for updating of %s: %+v", id, err)
+				}
+			}
+
+			metadata.SetID(id)
+
+			return nil
+		},
+	}
+}
+
+func (AzureAIServicesResource) Read() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 5 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.Cognitive.AccountsClient
+			env := metadata.Client.Account.Environment
+
+			state := AzureAIServicesResourceResourceModel{}
+			id, err := cognitiveservicesaccounts.ParseAccountID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			resp, err := client.AccountsGet(ctx, *id)
+			if err != nil {
+				if response.WasNotFound(resp.HttpResponse) {
+					return metadata.MarkAsGone(id)
+				}
+				return fmt.Errorf("retrieving %s: %+v", *id, err)
+			}
+
+			keys, err := client.AccountsListKeys(ctx, *id)
+			if err != nil {
+				// note for the resource we shouldn't gracefully fail since we have permission to CRUD it
+				return fmt.Errorf("listing the Keys for %s: %+v", id, err)
+			}
+
+			if model := keys.Model; model != nil {
+				state.PrimaryAccessKey = pointer.From(model.Key1)
+				state.SecondaryAccessKey = pointer.From(model.Key2)
+			}
+
+			state.Name = id.AccountName
+			state.ResourceGroupName = id.ResourceGroupName
+
+			if model := resp.Model; model != nil {
+				state.Location = location.NormalizeNilable(model.Location)
+				if sku := model.Sku; sku != nil {
+					state.SkuName = sku.Name
+				}
+
+				identityFlatten, err := identity.FlattenSystemAndUserAssignedMapToModel(model.Identity)
+				if err != nil {
+					return err
+				}
+				state.Identity = *identityFlatten
+
+				if props := model.Properties; props != nil {
+					state.Endpoint = pointer.From(props.Endpoint)
+					state.CustomSubdomainName = pointer.From(props.CustomSubDomainName)
+					state.NetworkACLs = flattenAzureAIServicesNetworkACLs(props.NetworkAcls)
+					state.Fqdns = pointer.From(props.AllowedFqdnList)
+
+					state.PublicNetworkAccess = string(*props.PublicNetworkAccess)
```

Comment:
Potential crash here 
```suggestion
					state.PublicNetworkAccess = string(pointer.From(props.PublicNetworkAccess))
```